src/assignment3/Assignment3.java

```diff
@@ -0,0 +1,67 @@
+package assignment3;
+
+import java.util.HashSet;
+import java.util.Set;
+
+public class Assignment3 {
+
+    private static void recursiveSearch(char[][] grid, boolean visited[][],
+            int width, int height, int startI, int startJ,
+            Dictionary dict, String prefix, Set<String> foundWords) {
+        
+        visited[startI][startJ] = true;
+        if (dict.isWord(prefix)) {
+            if (!foundWords.contains(prefix)) {

```

Comment:
This check is not really needed. You are using set to store results, so you can safely add the same value multiple times.